syntax = "proto3";

package app;

service AppService {
  rpc login(LoginRequest) returns (LoginResponse);
  rpc post(PostRequest) returns (PostResponse);
  rpc get(GetRequest) returns (GetResponse);
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  string status = 1;
  string token = 2;
}

message PostRequest {
  string token = 1;
  string type = 2;  // "add_inventory", "update_inventory", "query_ai"
  string data = 3;
}

message PostResponse {
  string status = 1;
  string result = 2;
}

message GetRequest {
  string token = 1;
  string type = 2;
  string params = 3;
}

message GetResponse {
  string status = 1;
  repeated string items = 2;
}
